smithy-cli/src/main/java/software/amazon/smithy/cli/commands/InitCommand.java

```diff
@@ -86,32 +90,100 @@ private int run(Arguments arguments, Env env) {
         Options options = arguments.getReceiver(Options.class);
         StandardOptions standardOptions = arguments.getReceiver(StandardOptions.class);
 
-        try {
-            Path root = Paths.get(".");
-            Path temp = Files.createTempDirectory("temp");
+        boolean isLocalRepo = isLocalRepo(options.repositoryUrl);
+        Path templateRepoDirPath = getTemplateRepoDirPath(options.repositoryUrl, isLocalRepo);
+
+        // If the cache directory does not exist, create it
+        if (!Files.exists(templateRepoDirPath)) {
```

Comment:
Does it make sense to add a param to allow user to invalidate the cache?

Reply:
`smithy clean` does that